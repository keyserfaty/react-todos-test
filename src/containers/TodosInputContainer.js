import React from 'react'
import TodosInput from '../components/TodosInput'

class TodosInputContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      inputText: '',
    };

    this.onInputChange = this.onInputChange.bind(this);
  }

  onInputChange(e) {
    this.setState({
      inputText: e.target.value,
    });
  }

  render() {
    const { inputText } = this.state;
    const { onInputChange } = this;
    const { onAddTodoClick } = this.props;

    return (
      <TodosInput inputText={inputText} onInputChange={onInputChange} onAddTodoClick={onAddTodoClick} />
    );
  }
}

export default TodosInputContainer
