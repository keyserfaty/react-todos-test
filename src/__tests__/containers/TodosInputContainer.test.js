import React from 'react'
import renderer from 'react-test-renderer'

import TodosInputContainer from '../../containers/TodosInputContainer'

it('should render a todos input component', () => {
  const tree = renderer.create(
    <TodosInputContainer />
  ).toJSON();

  expect(tree).toMatchSnapshot()
});
