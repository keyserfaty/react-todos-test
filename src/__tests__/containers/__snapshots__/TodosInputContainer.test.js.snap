exports[`test should render a todos input component 1`] = `
<div
  id="todos-input">
  <input
    onChange={[Function]}
    type="text"
    value="" />
  <button
    onClick={[Function]}>
    Add todo!
  </button>
</div>
`;
